### DISCUSSION TITLE
Please make "Hires. fix" available in img2img

### DISCUSSION THREAD

**Discussion by vihangasa14** [lllyasviel/stable-diffusion-webui-forge#758] [Ideas]
Hi guys and thanks for your amazing work, I have a request though : please make "Hires. fix" available in img2img too.

Sure, theoretically we should be able to manually replicate what it does in img2img, but the truth is that many of us fail to do so. "Hires. fix"'s second pass somehow makes pictures look so much better and we don't get to replicate this, even by using Loopback or Extra or Upscale or any settings. In help forums this is often asked.

If there is nothing keeping it from working in img2img, even if it's theoretically just an automation of a second pass, please just let us use it.

Thanks

**Comment by andy8992** [comment]
I was just going to make a topic about this. No matter what I do, the quality is nowhere near the original hiresfix button in txt2img

**Comment by ScionoicS** [comment]
hiresfix is just img2img done automatically. 

you could try using a resize by multiplier and put it through the loop back script. 

edit: i thought the loopback would multiply the size each loop but it only used the initial size.

**Comment by HMRMike** [comment]
Making sure to choose the same upscaler for img2img (look in the settings under "upscaling") as for hires fix, the differences seem very minor. Or maybe I'm not running into the cases which cause significant difference. 
512X512 upscaled 2X with 0.5 denoise, same seed, prompt, everything as identical as possible. NMKD SIAX upscaler. The most noticeable differences are with the leaf on the bottom left and the whiskers on the right.  With 0.2 denoising, the differences are smaller but still there.
![00154-1234](https://github.com/user-attachments/assets/da9afd79-c238-41bf-b09a-04686abff0c0)

Hires fix:
![00155-1234](https://github.com/user-attachments/assets/9e3de47a-a8a7-4181-9b19-e4e3144fb7cf)

img2img:
![00031-1234](https://github.com/user-attachments/assets/dd5034d5-063e-4cba-997c-6493587ae787)

Using Lanczos upscale the results are perfectly 1:1, this is with 0.5 denoising, but no differences were seen at 0.75 and so on.
![00159-1234](https://github.com/user-attachments/assets/7db50817-23e8-43e3-9d53-ffc419bcc060)
![00034-1234](https://github.com/user-attachments/assets/be008360-b244-46f6-affd-bdb785148008)

While I wonder what could be the cause of any differences at all with ESRGAN upscalers, I don't see the issue. Mostly we'll just need the upscaler setting to be available under img2img- you can add it using ![image](https://github.com/user-attachments/assets/f09c16e0-d5de-4bfa-9623-fe398ff52b63)
